model/vfs/vfsafero/impl.go

```diff
@@ -255,6 +245,85 @@ func (afs *aferoVFS) CreateFile(newdoc, olddoc *vfs.FileDoc, opts ...vfs.CreateO
 	}, nil
 }
 
+func (afs *aferoVFS) CopyFile(olddoc *vfs.FileDoc, newdoc *vfs.FileDoc) (err error) {
+	var newfile *aferoFileCreation
+	defer func() {
+		// XXX: we need to release the VFS lock before closing newfile as
+		// aferoFileCreation.Close requests its own lock.
+		// Therefore, this defer method needs to come before the afs.mu.Unlock
+		// deferred call.
+		if newfile == nil {
+			return
+		}
+		if cerr := newfile.Close(); cerr != nil && err == nil {
+			err = cerr
+		}
+	}()
+
+	if lockerr := afs.mu.Lock(); lockerr != nil {
+		return lockerr
+	}
+	defer afs.mu.Unlock()
+
+	newsize, maxsize, capsize, err := vfs.CheckAvailableDiskSpace(afs, newdoc)
+	if err != nil {
+		return err
+	}
+
+	newname := newdoc.DocName
+	exists, err := afs.Indexer.DirChildExists(newdoc.DirID, newname)
+	if err != nil {
+		return err
+	}
+	if exists {
+		newname = vfs.ConflictName(afs, newdoc.DirID, newname, true)
+		exists, err = afs.Indexer.DirChildExists(newdoc.DirID, newname)
+		if err != nil {
+			return err
+		}
+		if exists {
+			return os.ErrExist
+		}
+	}
+	newdoc.DocName = newname
+
+	f, err := afero.TempFile(afs.fs, "/", newdoc.DocName)
+	if err != nil {
+		return err
+	}
+	tmppath := path.Join("/", f.Name())
+
+	// XXX: we use the internal openFile method as we already have a VFS lock
+	content, err := afs.openFile(olddoc)
+	if err != nil {
+		return err
+	}
+	defer content.Close()
+
+	hash := md5.New()
+	extractor := vfs.NewMetaExtractor(newdoc)
+
+	newfile = &aferoFileCreation{
+		afs:     afs,
+		f:       f,
+		newdoc:  newdoc,
+		tmppath: tmppath,
+		w:       0,
+		size:    newsize,
+		maxsize: maxsize,
+		capsize: capsize,
+		hash:    hash,
+		meta:    extractor,
+	}
+
+	_, err = io.Copy(newfile, content)
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
The sequence:

```go
	if err != nil {
		return err
	}

	return nil
```

can be simplified to:

```go
	return err
```

Reply:
Totally. This part has changed though and the sequence is now relevant.